jni/src/dartjni.c

```diff
@@ -514,3 +516,38 @@ FFI_PLUGIN_EXPORT JNIEnv* GetJniEnv() {
   attach_thread();
   return jniEnv;
 }
+
+FFI_PLUGIN_EXPORT intptr_t InitDartApiDL(void* data) {
+  return Dart_InitializeApiDL(data);
+}
+
+JNIEXPORT void JNICALL
+Java_com_github_dart_1lang_jni_PortContinuation__1resumeWith(JNIEnv* env,
+                                                             jobject thiz,
+                                                             jlong port,
+                                                             jobject result) {
+  Dart_CObject dartPtr;
+  dartPtr.type = Dart_CObject_kInt64;
+  dartPtr.value.as_int64 = (jlong)((*env)->NewGlobalRef(env, result));
+  Dart_PostCObject_DL(port, &dartPtr);
+}
+
+// com.github.dart_lang.jni.PortContinuation
+jclass _c_PortContinuation = NULL;
+
+jmethodID _m_PortContinuation__ctor = NULL;
+FFI_PLUGIN_EXPORT
+JniResult PortContinuation__ctor(int64_t j) {
```

Comment:
`__ctor` is a slightly strange suffix. It is not coming from existing Java or C code right?

In the Dart ecosystem we prefer spelling out names instead of using abbreviations.

`PortContinuationConstructor` Or if you want to signal `PortContinuation` is a Java class, `PortContinuation_Constructor`?

Reply:
It's coming from the generated code from jnigen without any change actually. I can change the name.

Reply:
Ah, generated names from Java->Dart (dealing with overloading etc) is something we should revisit before going 1.0

Consider it moot for this PR then.